.ci/pytorch/build.sh

```diff
@@ -168,6 +164,15 @@ if [[ "${BUILD_ENVIRONMENT}" == *clang* ]]; then
   export CXX=clang++
 fi
 
+if [[ "$BUILD_ENVIRONMENT" == *-clang7-asan* ]]; then
+  export LDSHARED="clang --shared"
+  export USE_CUDA=0
+  export USE_ASAN=1
+  export USE_MKLDNN=0
```

Comment:
Because openmp can't be found by cmake when combining MKLDNN. I need more time to investigate. If you check build-asan.sh, MKLDNN is also disabled, someone encountered the case before.